xoto3/dynamodb/write_versioned/specify.py

```diff
@@ -0,0 +1,89 @@
+"""Low level API for specifying non-effect facts about the state of the database"""
+from typing import Optional
+
+from .ddb_api import table_name as _table_name
+from .keys import hashable_key, standard_key_attributes
+from .types import Item, ItemKey, TableNameOrResource, VersionedTransaction, _TableData
+
+
+def presume(
```

Comment:
Most of the other functions seem to make intuitive sense to me, I just want to make sure I'm understanding this one correctly. It's similar to get but it doesn't actually do any of the work but you can use it as if it was a get?

Reply:
to start off with, `presume` is 100% an optimization. You never _need_ to use it, and it will never affect the correctness of your transaction, just how long it might take to run and how much money will be spent in the process.

a `get` forces (eventually) a read from the table. `presume` says "if you haven't already read this from the table, hold off on that - I think I know what the value is." Because of the nature of how these transactions confirm their fetched values upon commit, if you turn out to be wrong about what you presumed, then your transaction will just retry. And if we already _know_ what the value of the item is, then we'll just ignore what you suggested and move on.

does that help?  i tried to say all of the above in more technical language in the actual docstrings.  Perhaps at some point we're going to want to surface those docstrings more directly in the README, though I'm averse to duplicating anything in that way.

Reply:
So even if you don't have a `presume` it'll force a retry regardless?

Reply:
from an implementation perspective, what happens is that we will always make sure that we knew the correct 'before' value for every item referenced in your transaction, and the transaction will retry until the value at the time of commit is the same as the value when your Python code actually executed.

this is the essential concept behind transactions - that you get to live in a world as though you were the only person talking to the database from the moment you started trying to do something until the moment you said you were done. In order to accomplish that, any situation where something appears to have changed will cause a fresh retry from the beginning.